### PR TITLE
feat: don't allow none folder items to be published

### PR DESCRIPTION
**Pull request by ReidyT**
- also improve the description components of the status button

Screenshot 2024-06-11 at 15 37 52
Screenshot 2024-06-11 at 15 37 31
Screenshot 2024-06-11 at 15 37 15
Screenshot 2024-06-11 at 15 36 51
Screenshot 2024-06-11 at 15 36 30
Screenshot 2024-06-11 at 15 34 41
Screenshot 2024-06-11 at 15 33 28
Screenshot 2024-06-11 at 15 31 36
image

### CODE REVIEW

File: cypress/e2e/item/publish/publishedItem.cy.ts [hunk 1]
```diff
@@ -340,4 +341,20 @@ describe('Public Item', () => {
       waitOnUnpublishItem(publicItem);
     });
   });
+
+  describe('Cannot Publish None Folder Item', () => {
```
Comment (pyphilia) [inline cypress/e2e/item/publish/publishedItem.cy.ts:345]:
```suggestion
  describe('Cannot Publish Non-Folder Item', () => {
```

File: src/components/hooks/usePublicationStatus.tsx [hunk 2]
```diff
@@ -83,6 +84,8 @@ const computePublicationStatus = ({
   switch (true) {
     case isPublishedChildren({ item, publishedEntry }):
       return PublicationStatus.PublishedChildren;
+    case itemTypeNotAllowedToBePublished(item.type):
```
Comment (pyphilia) [inline src/components/hooks/usePublicationStatus.tsx:87]: Same remark as in the backend. You can create your own function for the frontend.

File: src/langs/en.json [hunk 3]
```diff
@@ -186,6 +186,7 @@
   "LIBRARY_SETTINGS_PUBLISH_NOTIFICATIONS_LABEL": "Send email notifications to all co-editors",
   "LIBRARY_SETTINGS_PUBLISHED_STATUS": "This element is published. Anyone can access it and is available on Graasp Library, our public repository of learning ressources.",
   "LIBRARY_SETTINGS_CHILD_PUBLISHED_STATUS": "This element is part of a collection that is already published. Publishing and un-publishing can be done at the collection root.",
+  "LIBRARY_SETTINGS_TYPE_NOT_ALLOWED_STATUS": "This element can not be published. Only folders are allowed to be published on Graasp Library.",
```
Comment (pyphilia) [inline src/langs/en.json:189]: Should we have dynamic text for types? I think we have translations for types as well in common.

File: src/langs/en.json [hunk 4]
```diff
@@ -218,6 +219,7 @@
   "LIBRARY_SETTINGS_PUBLICATION_STATUS_INVALID": "Invalid",
   "LIBRARY_SETTINGS_PUBLICATION_STATUS_OUTDATED": "Outdated",
   "LIBRARY_SETTINGS_PUBLICATION_STATUS_UNPUBLISHED": "Unpublished",
+  "LIBRARY_SETTINGS_PUBLICATION_STATUS_TYPE_NOT_ALLOWED": "Item Type not allowed",
```
Comment (pyphilia) [inline src/langs/en.json:222]: I would have something more specific (file cannot be published, link cannot be published)

**Review by pyphilia — CHANGES_REQUESTED** [review]
Thanks for the changes! 
Actually I wonder if it wouldn't be better to hide the publication page completely for given types? Or just show a text, instead of providing the form on the left but it's unnecessary to fill?

File: src/langs/en.json (hunk 4 above)
Comment (ReidyT) [reply src/langs/en.json:222]: Really good idea !

File: src/langs/en.json (hunk 3 above)
Comment (ReidyT) [reply src/langs/en.json:189]: Yes it's an interesting point !

File: src/components/hooks/usePublicationStatus.tsx (hunk 2 above)
Comment (ReidyT) [reply src/components/hooks/usePublicationStatus.tsx]: As I wrote in the backend PR, for me it is not an issue to use a function from the SDK, maybe we can discuss in the backend's PR ?

**Review by pyphilia — APPROVED** [review]

**Pull request merged**